go/vt/tabletmanager/reparent.go

```diff
@@ -18,14 +23,191 @@ import (
 	"golang.org/x/net/context"
 )
 
+var (
+	fastReparent = flag.Bool("fast_external_reparent", false, "Skip updating of fields in topology that aren't needed if all MySQL reparents are done by an external tool, instead of by Vitess directly.")
+
+	finalizeReparentTimeout = flag.Duration("finalize_external_reparent_timeout", 10*time.Second, "Timeout for the finalize stage of a fast external reparent reconciliation.")
+)
+
+// fastTabletExternallyReparented completely replaces TabletExternallyReparented
+// if the -fast_external_reparent flag is specified.
+func (agent *ActionAgent) fastTabletExternallyReparented(ctx context.Context, externalID string) (err error) {
+	tablet := agent.Tablet()
+
+	log.Infof("using fast TabletExternallyReparented")
```

Comment:
Good catch :) I wanted the word "fast" to be prominent so we know at a glance which version is running. I've incorporated that into the other existing log lines and removed this unconditional one.